src/core/TriangleSplitter.js

```diff
@@ -189,7 +212,37 @@ export class TriangleSplitter {
 				// so we can use that information to determine whether to split later.
 				const startDist = plane.distanceToPoint( _edge.start );
 				const endDist = plane.distanceToPoint( _edge.end );
-				if ( Math.abs( startDist ) < COPLANAR_EPSILON && Math.abs( endDist ) < COPLANAR_EPSILON ) {
+
+				// we scale COPLANAR_EPSILON since a very small edge leads to more uncertainty in the true plane position:
+				// see https://github.com/gkjohnson/three-bvh-csg/issues/199#issuecomment-1986287165
+				let coPlanarEpsilonStart = COPLANAR_EPSILON *	Math.max( 1, 0.5 * _edge.start.distanceTo( planeCenter ) / planeEdgeSize );
+				let coPlanarEpsilonEnd = COPLANAR_EPSILON *	Math.max( 1, 0.5 * _edge.end.distanceTo( planeCenter ) / planeEdgeSize );
+
+				coPlanarEpsilonStart = Math.min( coPlanarEpsilonStart, COPLANAR_EPSILON_MAX );
+				coPlanarEpsilonEnd = Math.min( coPlanarEpsilonEnd, COPLANAR_EPSILON_MAX );
+
+
+				const isStartCoplanar = ( Math.abs( startDist ) < coPlanarEpsilonStart );
+				const isEndCoplanar = ( Math.abs( endDist ) < coPlanarEpsilonEnd );
+
+				// reprojection:
+				// if we estimate that a point belongs to the plane
+				// we force it to belongs to the plane
+				// I cannot explain exactly why it works but it looks that it works
```

Comment:
This bit makes me uncomfortable still, I think. It's something that could fix your case and break others. Likely what's happening is that it's shifting vertices which helps change the subsequent calculations just enough such that it produces a different result and isn't culled. But this isn't stable in the sense the order in which triangles are clipped would change the result. And you'd get a similar effect by just randomly jittering the vertices by a small amount, I believe.

Instead of modifying the original triangle vertices such that it impacts subsequent clipping calculations have you tried just project the edge points in the local scope for this calculation to see if that solves the issue? Ie do something like this:

```js
if ( isStartCoplanar ) {

	plane.projectPoint( _edge.start, _edge.start );

}

if ( isEndCoplanar ) {

	plane.projectPoint( _edge.end, _edge.end );

}
```

Then we're ensuring that at least the edge we're using is on the other triangle plane.